packages/alfa-rectangle/src/rectangle.ts

```diff
@@ -86,28 +144,76 @@ export class Rectangle
     );
   }
 
-  equals(value: this): boolean;
-  equals(value: unknown): value is this;
-  equals(value: unknown): boolean {
+  public union(other: Rectangle): Rectangle {
+    if (this.contains(other)) {
+      return this;
+    }
+
+    if (other.contains(this)) {
+      return other;
+    }
```

Comment:
**Question:** Are these early escapes really needed? It seems that the computations below should give the same result anyway, and these don't save that much time since `.contains` itself makes more or less the same comparisons.

Reply:
Right. My reason for doing it was that it reuses the existing object rather than allocating a new one which would have to be garbage collected etc. I don't have a strong opinion though on whether it's worth the extra lines of code and potential confusion.